apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.js

```diff
@@ -97,27 +97,36 @@ function processSubmission(req, res, callback) {
 }
 
 router.post('/', function (req, res, next) {
-  if (res.locals.assessment.type !== 'Exam') return next();
-
   if (!res.locals.authz_result.authorized_edit) {
     return next(error.make(403, 'Not authorized', res.locals));
   }
 
   if (req.body.__action === 'grade' || req.body.__action === 'save') {
-    if (res.locals.authz_result.time_limit_expired) {
-      return next(
-        error.make(403, 'time limit is expired, please go back and finish your assessment'),
-      );
-    }
-    if (req.body.__action === 'grade' && !res.locals.assessment.allow_real_time_grading) {
-      next(error.make(403, 'Real-time grading is not allowed for this assessment'));
-      return;
+    if (res.locals.assessment.type === 'Exam') {
+      if (res.locals.authz_result.time_limit_expired) {
+        return next(
+          error.make(403, 'time limit is expired, please go back and finish your assessment'),
```

Comment:
Nit, capital error message for consistency.

```suggestion
          error.make(403, 'Time limit is expired, please go back and finish your assessment'),
```